### ISSUE TITLE
Py2neo RelatedTo/From cause circular import errors

### ISSUE THREAD

**Issue by digerati9** [technige/py2neo#740]
We have a graph with
(a)-[:LINKS_TO]->(b)

class a(GraphObject):
   …
   links_to = RelatedTo(b, “LINKS_TO”)

class b(GraphObject):
 …
 links_from = RelatedFrom(a, “LINKS_TO”)

Note: using the objects a and b here not ‘a’ and ‘b’.

This works perfectly in class b; I can jump through the relationship and work with the a object perfectly.

With class b, it tells me a cannot be found because the class is not yet defined.

Python is unable to handle these circular references.  

We are experimenting with using ‘a’ in the RelatedTo definition, however, at first try, it doesn’t return the GraphObject the same way (we get an empty return).

The py2neo documentation example is here: https://py2neo.org/v4/ogm.html

It uses quotes in the first class and objects in the 2nd.  Experimenting with this setup today, but out relationships are more complex so examples of successful implementation would be helpful.

This MUST come up in nearly all Py2neo OGM implementations since you would want to reference the related objects from nearly every class.

What is a good solution to this?

**Comment by technige** [comment]
The string variants of the class names are intended to help work around exactly this problem. If class A references class B before B is defined, the name of B will need to be passed in quotes.

This is precisely what the example you reference is doing:
```
class Movie(GraphObject):
    __primarykey__ = "title"

    title = Property()
    tag_line = Property("tagline")
    released = Property()

    actors = RelatedFrom("Person", "ACTED_IN")
    directors = RelatedFrom("Person", "DIRECTED")
    producers = RelatedFrom("Person", "PRODUCED")


class Person(GraphObject):
    __primarykey__ = "name"

    name = Property()
    born = Property()

    acted_in = RelatedTo(Movie)
    directed = RelatedTo(Movie)
    produced = RelatedTo(Movie)
```

Here, the `Person` class does not exist when the `Movie` class is being defined. Therefore, a direct reference to `Person` would fail. Putting the name of the method in quotes causes the class to be lazily resolved after the entire module has loaded.

On the other hand, the `Person` class itself can of course directly reference the `Movie` class as the latter will already have been defined.

**Comment by digerati9** [comment]
Thank you @technige!  We discovered this after more testing.  

Is there any advantage to using the class directly (as in "acted_in = RelatedTo(Movie)") rather than always using the quotes?  The code looks like it checks if the class is imported and if not imports at that point.  

We discovered another interesting issue.   This "actors = RelatedFrom("Person", "ACTED_IN")" was not working for us.  We discovered that the "Person" portion is looking for the class name starting from the  root of the application.  

Our classes here were stored at root/data/classname.py and to resolve we had to use that path as the class:
"products = RelatedTo('data.product.Product', 'MARKETS')"
